### PR TITLE
Fixes coverage reporting and some documentation

### PR DESCRIPTION
**Pull request by jfboismenu**
Code Coverage upload broke almost as soon as it was introduced. We were relying on token-less uploads to codecov, but apparently the feature broke on their end and now it does not work anymore, so we're now going to always pass it in as a precaution.

The pipeline will now expected a secret named `codecov.token` to exist on the pipeline and use that to initialize `CODECOV_TOKEN`.

I've also enabled code coverage upload on all variants of the build now that everything looks like it is working and we are now getting coverage of custom code paths based on the OS and Python version.

I've also take the time to document the steps required to add a pipeline on Azure Pipelines.

I've renamed the `templates` folder to `internal` to better reflect the non-reusability of these files.

Finally, I've jolted down some thoughts about how a future unit-tests.yml file could be structured. We're going to want to move in this direction at some point, as we're not guaranteed that all Toolkit bundles will always be compatible with all OSes, even though they should always strive to be, and we should always be able to turn off certain builds if something needs to be skipped.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,3 +104,27 @@ jobs:
 If you need to use an in-development version of these tools from a branch, you can update the `ref` in `repositories` section to point to a different branch.
 
 The `build-pipeline.yml` template has a few parameters. You can find out more about them in [build-pipeline.yml](https://github.com/shotgunsoftware/tk-ci-tools/blob/master/build-pipeline.yml).
+
```
Comment (jfboismenu) [inline README.md:107]: You can preview the documentation [here](https://github.com/shotgunsoftware/tk-ci-tools/blob/478ecd2afcfccc2af109a2a24c9f7debed19bbe7/README.md#create-a-build-in-azure-pipelines)

File: README.md [hunk 2]
```diff
@@ -108,3 +104,27 @@ jobs:
...
+### Create a build in Azure Pipelines
+
+Once you've added your `azure-pipelines.yml` file to your repository, it's time to add the repository to the Azure Pipelines web service.
+
+1. Go to the [Shotgun Ecosystem Azure Pipelines](https://dev.azure.com/shotgun-ecosystem/Toolkit/_build) page.
+2. Select any pipeline from the list and clone it.
+   ![cloning_build](resources/cloning_build.png)
+3. In the YAML tab, update the build info
+3.1 The name can be updated from the right-hand side of the page. You must name the build after the name of the repository.
```
Comment (willis102) [inline README.md]: Github's markdown isn't parsing this correctly:

Screen Shot 2019-11-04 at 11 43 10 AM

**Review by willis102 — APPROVED** [review]
The documentation is great.  So many screenshots ❤️

Only thing I was was that it looks like Github's markdown parser doesn't like the sub-enumeration, see below.

File: README.md (hunk 2 above)
Comment (jfboismenu) [reply README.md]: Ah, I had not noticed! I'll fix it right away.

**Comment by jfboismenu** [comment]
Since the code review I've slipped in a totally unrelated feature, which we'll need now that we are porting applications to Python 3. Most of these do not have any kind of tests and there needs to be a way to skip tests for these repos.

**Pull request merged**